hatch_build.py

```diff
@@ -22,13 +22,32 @@ class CustomBuildHook(BuildHookInterface):
     def local_cffi_license(self) -> str:
         return os.path.join(self.root, self.LICENSE_NAME)
 
+    @staticmethod
+    def get_cffi_distribution_license_files() -> list[PackagePath]:
+        license_files = []
+
+        dist_files = distribution("cffi").files or []
+        for f in dist_files:
+            if f.name != "LICENSE":
+                continue
+
+            current = f.parent
+            while current and not current.name.endswith(".dist-info"):
+                current = current.parent
+                if current is None or len(current.parts) == 0:
+                    break
```

Comment:
Not entirely equivalent, but doesn't it do the job?
```suggestion
            current = f.parent
            while current and current.parts and not current.name.endswith(".dist-info"):
                current = current.parent
```

Reply:
`current` is always "truthly" (`PackagePath('.')`), so it can be removed

```python
while current.parts and not current.name.endswith(".dist-info"):
```